### DISCUSSION TITLE
Provide an option to submit a parameter only once during a full script execute

### DISCUSSION THREAD

**Discussion by Mingo1111** [dbeaver/dbeaver#36130] [Ideas]
When executing a full script (ALT+X) with multiple queries, it would be helpful to have the parameter arguments universal across all executing queries, instead of prompting for a new set of arguments with each individual query. 

Here is a current functionality example using ALT+X:
```
SELECT * FROM user
WHERE uid = :UID;

/* PROMPT 1 (appears after above query) */
/* UID PROMPT 1 = 'TEST' */

SELECT * FROM purchase
WHERE uid = :UID
  AND calendar_year = :YEAR;

/* PROMPT 2 */
/* UID PROMPT 2 = 'TEST' AND
 * YEAR PROMPT 1 = '2024'
 */

``` 

Here is the desired functionality example:
```
SELECT * FROM user
WHERE uid = :UID;

/* UNIVERSAL PROMPT (PARAMETER SHARING OPTION SELECTED) */
/* UID PROMPT 1 = 'TEST' AND
 * YEAR PROMPT 1 = 2024
*/

SELECT * FROM purchase
WHERE uid = :UID
  AND calendar_year = :YEAR;

/* PROMPT 2 not needed */

``` 

Ideally the user would check a box to toggle this at the time of the first prompt which looks ahead to the full execution section and finds each **distinct** parameter name, and prompts for them all at the same time. Once the arguments are all input and the user hits OK, it would then execute all queries and apply the arguments to each named parameter during this execution where they share the same name, without any further prompts.

I could see it as just a checkbox option at the time of the first prompt, but a setting under Editors > SQL Editor > SQL Processing > Parameters section could be helpful also.

I recognize this can kind of be achieved by using variables, however, when you switch from query to query (ALT+X or CTRL+ENTER) to another query that contains the same parameter name (even in a different sql file) that value is already stored as the "default" for the next execution. So my thought here is to just have that already stored value be applied automatically to the all of queries when they are executed at the same time. The benefit here is being able to skip the additional prompts when using the same parameter name many times throughout the same execution.

**Comment by TB-HunBoka** [comment]
I see what you would like to achieve, but here is how I usually handle this.
If you fine with having some extra rows, then you can achive this by a `@set` command. This will create the variable as you mentioned, however it will also replace stored variables, so having the same variable is not a problem.

Example:
```
@SET UID = 5
@SET YEAR = 2024

SELECT * FROM user
WHERE uid = :UID;

SELECT * FROM purchase
WHERE uid = :UID
  AND calendar_year = :YEAR;
```

**Reply by Mingo1111** [reply]
I'll have to give this a try, I thought you had to change them to $ variable $ notation for this to work with the set command.
